### PR TITLE
Add basic modal component (Issue #56)

### PR DESCRIPTION
**Pull request by Dmitri-2**
Adds a basic version of the modal component. This component will still need additional updates to align with the functionality of the main Flowbite modal component. This first version implements the functions: 
- show
- hide 
- toggle

### CODE REVIEW

File: docs/components/modal/examples/ModalExample.vue [hunk 1]
```diff
@@ -1,8 +1,72 @@
...
+    default() {
+      return []
+    },
+  },
+  popup: {
+    type: Boolean,
+    default: false,
+  },
+  position: {
+    type: String, // 'bottom-left' | 'bottom-right' | 'bottom-center' | 'top-left' | 'top-center' | 'top-right' | 'center-left' | 'center' | 'center-right'
```
Comment (cogor) [inline docs/components/modal/examples/ModalExample.vue:56]: You need to extract this to type like [here](https://github.com/themesberg/flowbite-vue/blob/main/src/components/Avatar/types.ts)

File: docs/components/modal/examples/ModalExample.vue [hunk 2]
```diff
@@ -1,8 +1,72 @@
...
+    default: 'center',
+  },
+  show: {
+    type: Boolean,
+    default: false,
+  },
+  size: {
+    type: String, // 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | '4xl' | '5xl' | '6xl' | '7xl'
```
Comment (cogor) [inline docs/components/modal/examples/ModalExample.vue:64]: You need to extract this to type

File: src/components/Modal/Modal.vue [hunk 3]
```diff
@@ -1,37 +1,34 @@
...
-          </p>
-        </div>
-        <!-- Modal footer -->
-        <div class="flex items-center p-6 space-x-2 rounded-b border-t border-gray-200 dark:border-gray-600">
-          <button data-modal-toggle="defaultModal" type="button" class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800">I accept</button>
-          <button data-modal-toggle="defaultModal" type="button" class="text-gray-500 bg-white hover:bg-gray-100 focus:ring-4 focus:outline-none focus:ring-blue-300 rounded-lg border border-gray-200 text-sm font-medium px-5 py-2.5 hover:text-gray-900 focus:z-10 dark:bg-gray-700 dark:text-gray-300 dark:border-gray-500 dark:hover:text-white dark:hover:bg-gray-600 dark:focus:ring-gray-600">Decline</button>
+  <div>
+    <slot name="trigger" :show="showModal" :hide="hideModal" :toggle="toggleModal"></slot>
```
Comment (cogor) [inline src/components/Modal/Modal.vue:3]: You can use [self-closing component](https://v2.vuejs.org/v2/style-guide/?redirect=true#Self-closing-components-strongly-recommended)

File: src/components/Modal/Modal.vue [hunk 4]
```diff
@@ -1,37 +1,34 @@
...
+    <div v-if="!isHidden" tabindex="-1" :aria-hidden="isHidden ? 'true' : 'false'"
+         class="overflow-y-auto overflow-x-hidden fixed top-0 right-0 left-0 z-50 w-full md:inset-0 h-modal md:h-full justify-center items-center flex">
+      <div class="relative p-4 w-full h-full md:h-auto"
+           :class="`${modalSizeClasses[size]}`">
+        <!-- Modal content -->
+        <div class="relative bg-white rounded-lg shadow dark:bg-gray-700">
+          <!-- Modal header -->
+          <div class="p-4 rounded-t"
+               :class="$slots.header ? 'border-b' : ''">
+            <slot name="header" :show="showModal" :hide="hideModal" :toggle="toggleModal"></slot>
```
Comment (cogor) [inline src/components/Modal/Modal.vue:14]:
```suggestion
            <slot name="header" :show="showModal" :hide="hideModal" :toggle="toggleModal" />
```

File: src/components/Modal/Modal.vue (hunk 3 above)
Comment (cogor) [inline src/components/Modal/Modal.vue:3]:
```suggestion
    <slot name="trigger" :show="showModal" :hide="hideModal" :toggle="toggleModal" />
```

File: src/components/Modal/Modal.vue [hunk 5]
```diff
@@ -1,37 +1,34 @@
...
+          </div>
+          <!-- Modal body -->
+          <div class="p-6" :class="$slots.header ? '' : 'pt-0'">
+            <slot name="body" :show="showModal" :hide="hideModal" :toggle="toggleModal"></slot>
```
Comment (cogor) [inline src/components/Modal/Modal.vue:18]:
```suggestion
            <slot name="body" :show="showModal" :hide="hideModal" :toggle="toggleModal" />
```

File: src/components/Modal/Modal.vue [hunk 6]
```diff
@@ -1,37 +1,34 @@
...
+          </div>
+          <!-- Modal footer -->
+          <div v-if="$slots.footer" class="p-6 rounded-b border-gray-200">
+            <slot name="footer"  :show="showModal" :hide="hideModal" :toggle="toggleModal"/>
```
Comment (cogor) [inline src/components/Modal/Modal.vue:22]:
```suggestion
            <slot name="footer"  :show="showModal" :hide="hideModal" :toggle="toggleModal" />
```

File: docs/components/modal/examples/ModalSizeExample.vue [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<template>
+  <div class="vp-raw flex justify-between">
+    <span>
+      <ModalExample size="xs" trigger-text="XS Modal"/>
```
Comment (cogor) [inline docs/components/modal/examples/ModalSizeExample.vue:4]:
```suggestion
      <ModalExample size="xs" trigger-text="XS Modal" />
```

File: docs/components/modal/examples/ModalSizeExample.vue [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    </span>
+    <span>
+      <ModalExample size="md" trigger-text="MD Modal"/>
```
Comment (cogor) [inline docs/components/modal/examples/ModalSizeExample.vue:7]:
```suggestion
      <ModalExample size="md" trigger-text="MD Modal" />
```

File: docs/components/modal/examples/ModalSizeExample.vue [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    </span>
+    <span>
+      <ModalExample size="xl" trigger-text="XL Modal"/>
```
Comment (cogor) [inline docs/components/modal/examples/ModalSizeExample.vue:10]:
```suggestion
      <ModalExample size="xl" trigger-text="XL Modal" />
```

File: docs/components/modal/examples/ModalSizeExample.vue [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    </span>
+    <span>
+      <ModalExample size="5xl" trigger-text="5XL Modal"/>
```
Comment (cogor) [inline docs/components/modal/examples/ModalSizeExample.vue:13]:
```suggestion
      <ModalExample size="5xl" trigger-text="5XL Modal" />
```

**Review by cogor — CHANGES_REQUESTED** [review]

**Comment by cogor** [comment]
Thanks for your PR, can you fix issues, which I found?

**Comment by Dmitri-2** [comment]
Yes! Sorry about the delay – PR is now updated.

**Comment by cogor** [comment]
Thank you!

**Pull request merged**